resources/js/store/index.ts

```diff
@@ -0,0 +1,40 @@
+import Vue from 'vue';
+import Vuex, {Store} from 'vuex';
+import { Inertia } from '@inertiajs/inertia';
+import mutations from './mutations';
+
+export function getInitialState() {
+    return {
+        loading: false,
+        lastSearchedIds: ''
+    }
+}
+
+export function createStore(): Store<{loading: boolean}>{
```

Comment:
```suggestion
export function createStore(): Store<RootState>{
```

Reply:
oh weird. i dont remember adding that. maybe it was automatically added.